### ISSUE TITLE
RKLLM_INFER_GET_LOGITS 推理结果输出漏词疑问

### ISSUE THREAD

**Issue by wohaiaini** [airockchip/rknn-llm#245]
假设prefill 有21个token，generate有12个token：
```
I rkllm: --------------------------------------------------------------------------------------
I rkllm:  Stage         Total Time (ms)  Tokens    Time per Token (ms)      Tokens per Second
I rkllm: --------------------------------------------------------------------------------------
I rkllm:  Init          1528.08          /         /                        /
I rkllm:  Prefill       373.42           21        17.78                    56.24
I rkllm:  Generate      1294.38          12        107.86                   9.27
I rkllm: --------------------------------------------------------------------------------------
```
1. 通过 RKLLM_INFER_GET_LOGITS 模式获得了 logits 的结果输出，从输出结果看，有两个部分：
    a. prefill部分（vocab:151674, tokens:21）一次性输出21 token
    b. generate部分（vocab:151674, tokens:1）分12次输出1token
    **想请问上述理解是否正确？**

2. 通过编写logits之后处理流程发现，**第一个字符无法获得输出**：
    a. 文本模式输出结果为：**The** sum of 1 and 2 is 3.
    b. 通过logits方式输出结果为：sum of 1 and 2 is 3.
    c. 不知道为啥**少了个The？**，实际token数显示也是12.

**Comment by waydong** [comment]
你好，因为prefill算完后就能出一个token，decode实际推理了11次。

**Comment by wohaiaini** [comment]
那我怎么处理才能把第一个 token（比如"The"）输出呢？
直接用prefill预测成了pricing，而不是The，因该是错了
**pricing** sum of 1 and 2 is 3.

**Comment by waydong** [comment]
prefill 假设维度是 21*151674，理论对最后一维的151674做采样就是输出的第一个token。

**Comment by wohaiaini** [comment]
简单点说，第一个Token是算完prefill之后出的，那么该token是不是也应该有对应的logits？

> 你好，因为prefill算完后就能出一个token，decode实际推理了11次。

**Comment by waydong** [comment]
> 简单点说，第一个Token是算完prefill之后出的，那么该token是不是也应该有对应的logits？
> 
> > 你好，因为prefill算完后就能出一个token，decode实际推理了11次。

它的logits就是prefill的logits的最后一维

**Comment by wohaiaini** [comment]
好的，谢谢！

**Comment by wohaiaini** [comment]
另外问一下，这个是不是只有prompt完全匹配和前缀匹配才能复用，有具体的策略或算法吗？
